src/components/secrets/Secret.tsx

```diff
@@ -290,6 +297,8 @@ export const SecretForm: React.FC<SecretFormProps> = function (props) {
     }
 
     async function handleSubmit(e) {
+        const secretNameRegex = new RegExp(/^[-.a-zA-Z0-9]+$/);
```

Comment:
regex is wrong... also extract out regex.. also there is cap on number of characters that can be used

Reply:
character limit is 253 for secret as well as for configmap name

Reply:
added validation for 253 characters